rag_experiment_accelerator/evaluation/plain_metrics.py

```diff
@@ -1,153 +1,163 @@
-from fuzzywuzzy import fuzz
-
-algorithms = textdistance.algorithms
+from rapidfuzz import fuzz
+from rapidfuzz import distance as d
+from textdistance import algorithms as alg
 
 
 # https://huggingface.co/spaces/evaluate-metric/bleu
-def bleu(predictions, references):
-    bleu = evaluate.load("bleu")
-
-    results = bleu.compute(predictions=predictions, references=references, max_order=2)
-    # multiplying by 100 to maintain consistency with previous implementation
-    return results["bleu"] * 100
-
-
-def fuzzy(doc1, doc2):
+def bleu(predictions: list[str], references: list[str]) -> float:
     """
-    Calculates the fuzzy score between two documents.
+    Computes the BLEU score between a list of candidate translations and a list of reference translations.
 
-    Parameters:
-        doc1 (str): The first document to compare.
-        doc2 (str): The second document to compare.
+    Args:
+        predictions (list): A list of candidate translations.
+        references (list): A list of reference translations.
 
     Returns:
-        int: The fuzzy score between the two documents.
+        float: The BLEU score between the candidate and reference translations.
     """
-    differences = fuzzy_compare_values(doc1, doc2)
+    bleu = evaluate.load("bleu")
 
-    return int(sum(differences) / len(differences))
+    results = bleu.compute(predictions=predictions, references=references, max_order=2)
+    # multiplying by 100 to maintain consistency with previous implementation
+    return results["bleu"] * 100
 
 
-def fuzzy_compare_values(value1, value2) -> list[float]:
+def fuzzy(str1: str, str2: str, match_type: str = "token_set_ratio") -> float:
     """
-    Compares two values using fuzzy string matching and appends the similarity score to a list of differences.
+    Compares two strings using fuzzy string matching and returns a similarity score.
 
     Args:
-        value1 (str): The first value to compare.
-        value2 (str): The second value to compare.
+        str1 (str): The first string to compare.
+        str2 (str): The second string to compare.
+        match_type (str): The type of fuzzy string matching to use. Options include:
+            - 'ratio'
+            - 'token_set_ratio'
+            - 'token_sort_ratio'
+            - 'partial_ratio'
+            - 'partial_token_sort_ratio'
+            - 'partial_token_set_ratio'
+            - 'WRatio'
+            - 'QRatio'
 
     Returns:
         A list of the similarity scores.
+
+    Raises:
+        ValueError: If the match type is not recognized.
     """
-    similarity_score = [fuzz.token_set_ratio(value1, value2)]
+    try:
+        fuzzy_match_fn = getattr(fuzz, match_type)
+    except AttributeError:
+        raise ValueError(f"Match type '{match_type}' is not recognized.")
+
+    similarity_score = fuzzy_match_fn(str1, str2)
     return similarity_score
 
 
-def levenshtein(value1, value2):
+def levenshtein(str1: str, str2: str) -> int:
     """
```

Comment:
Severity: Major - The `levenshtein` function's return type hint should be `float` instead of `int` to match the actual return type.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.